src/megachatapi_impl.cpp

```diff
@@ -5759,21 +5778,46 @@ MegaChatSessionHandler::~MegaChatSessionHandler()
 
 void MegaChatSessionHandler::onSessStateChange(uint8_t newState)
 {
-    if (rtcModule::ISession::kStateInProgress == newState)
+    switch (newState)
     {
-        MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
-        chatCall->setRemoteAudioVideoFlags(session->receivedAv());
-        API_LOG_INFO("Initial remote audio/video flags. ChatId: %s, callid: %s, AV: %s",
-                     Id(chatCall->getChatid()).toString().c_str(),
-                     Id(chatCall->getId()).toString().c_str(),
-                     session->receivedAv().toString().c_str());
+        case rtcModule::ISession::kStateWaitSdpOffer:
+        case rtcModule::ISession::kStateWaitSdpAnswer:
+        {
+            MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
+            chatCall->setSessionStatus(MegaChatCall::SESSION_STATUS_INITIAL, session->peer());
+            megaChatApi->fireOnChatCallUpdate(chatCall);
+            break;
+        }
+        case rtcModule::ISession::kStateInProgress:
+        {
+            MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
+            chatCall->setRemoteAudioVideoFlags(session->receivedAv());
+            API_LOG_INFO("Initial remote audio/video flags. ChatId: %s, callid: %s, AV: %s",
+                         Id(chatCall->getChatid()).toString().c_str(),
+                         Id(chatCall->getId()).toString().c_str(),
+                         session->receivedAv().toString().c_str());
 
-        megaChatApi->fireOnChatCallUpdate(chatCall);
+            chatCall->setSessionStatus(MegaChatCall::SESSION_STATUS_IN_PROGRESS, session->peer());
+
+            megaChatApi->fireOnChatCallUpdate(chatCall);
+            break;
+        }
+        case rtcModule::ISession::kStateDestroyed:
```

Comment:
Maybe we should fall-through here for `kStateTerminating`??

Reply:
Not really, better to only notify one of the events. But avoid to notify when the call itself is being terminated.
Done!